### ISSUE TITLE
[Feature Request] Fetch All Articles From A User via API

### ISSUE THREAD

**Issue by sleithdylan** [codu-code/codu#512]
## Context

This would be a more **"Good to have"** feature but what if for example, someone wants to display the articles they have written on Codú on their own personal website?

We could have the following endpoints:

- `/api/articles?username=dylansleith` (Used to filter and return all articles by username)
- `/api/articles/dylansleith/this-is-the-slug-frsj8g_dk` (Used to return all information from the article e.g. name, text, tags, slug, etc)

## Expected Behavior

There will be API Endpoints which will allow people to fetch and display blog articles on their own website.

## Current Behavior

There are no API Endpoints to fetch and display blog articles on their own website.

**Comment by NiallJoeMaher** [comment]
This could be good! I might protect it with an API key initially to stop us from getting spammed. Great idea. I think @kera-cudmore had suggested something similar.

**Comment by sleithdylan** [comment]
> This could be good! I might protect it with an API key initially to stop us from getting spammed. Great idea. I think @kera-cudmore had suggested something similar.

Yeah, I remember the chat we had in Discord about a month ago about this, I mentioned that I was using the **dev.to API** to pull the blog posts into my portfolio currently. Another feature that dev.to doesn't have which would definitely want me to use Codú API more if we were able to change the slug on Codú. 

I have posted a few articles already on **dev.to** but then decided to change the title of my blog post but when I changed it, it didn't change the slug, it just kept the old title which I had on the post as the slug. It would be cool if we could change it so that could be another cool feature?
